info-compiler/src/main/java/com/google/sps/webcrawler/WebCrawler.java

```diff
@@ -186,12 +198,37 @@ public String handleResponse(final HttpResponse response) throws IOException {
     }
   }
 
+  /** 
+   * For testing purposes.
+   */
+  Optional<NewsArticle> scrapeAndExtractHtml(URL url, Grant grant) {
```

Comment:
I would refactor the other scrapeAndExtractHtml to use as much of this as possible since this is the function actually being tested.